### PR TITLE
Split course and progress in seperate API. Document Course cmi5 format

### PR DESCRIPTION
**Pull request by mcginkel**

### CODE REVIEW

**Comment by mcginkel** [comment]
split course API in to course and progress
Add cmi5 course documentation
add reference xsd for coursestructure and extentions

**Review by edwinverwoerd — APPROVED** [review]

File: reference/course.v1.yaml [hunk 1]
```diff
@@ -90,9 +98,11 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/StreamCourse'
+                $ref: '#/components/schemas/Course'
       operationId: get-stream-course
-      description: Get a specific course by stream ID - returns basic metadata.
+      description: |-
+        Get a specific course by stream ID - returns basic metadata.
```
Comment (hhaarsma) [inline reference/course.v1.yaml:104]: Should this be course ID?

File: reference/course.v1.yaml [hunk 2]
```diff
@@ -90,9 +98,11 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/StreamCourse'
+                $ref: '#/components/schemas/Course'
       operationId: get-stream-course
```
Comment (hhaarsma) [inline reference/course.v1.yaml:102]: Should this be get-course?

File: reference/course.v1.yaml [hunk 3]
```diff
@@ -90,9 +98,11 @@ paths:
...
-      description: Get a specific course by stream ID - returns basic metadata.
+      description: |-
+        Get a specific course by stream ID - returns basic metadata.
+        Usually the course is found via the catalog, each product has a list of courses that is available for a student/teacher when that product is bought. This call is used to get more infomation, can also be used to see if there are any new versions of this course published
   '/course/{id}/cmi5e-course':
```
Comment (hhaarsma) [inline reference/course.v1.yaml:106]: wondering whether the -course has any added value, it is already in the context of course/{id}

File: reference/course.v1.yaml [hunk 4]
```diff
@@ -111,163 +121,118 @@ paths:
...
-      operationId: get-user-statements
-      description: Get statements for a specific user
+                type: object
+                properties: {}
+              examples:
+                example-1:
+                  value: {}
+      operationId: get-course-cmi5
+      description: |-
+        Get the meta data of a specific course by stream ID -
```
Comment (hhaarsma) [inline reference/course.v1.yaml:196]: should this be course ID?

File: reference/course.v1.yaml [hunk 5]
```diff
@@ -111,163 +121,118 @@ paths:
...
+        Get the meta data of a specific course by stream ID - 
+        The meta data contains information abou the course, including which chapters, paragraphs and assignable learning units are available, including links to the publishers playing envirionment so the student start that unit.
+        The LMS can use this to personalise the course per student and add or skip assignable units. This structure and information is also the base for exchanging progress.
+        NOTE: the course content itself is not part of the specification, it describes the meta information and how to navigate to the right point in the course, but not the course itself.
+
+        The CMI5 standard also provides a protocol to launch assignable units in a browser and get progress on thse units. This is not part of the SEM project. The format is used to exchange meta information, not to enforce runtime. For the moment it is expected that the url's in the content work without a need to setup sessions etc. each publisher is responsible for creating url's in such a way that single signof is enforced. example: https://toegang.noordhoff.nl/8717927999997?deepLinkId=e19ffbe3-2411-4d1f-8aa6-14fc43c2142a
```
Comment (hhaarsma) [inline reference/course.v1.yaml:201]: Do we need to state that Single Sign On between parties is supported and the mechanism (to my knowledge this is now the standard in the Dutch Secondary Education Eco-system; but might be good to state it explicit)

File: reference/course.v1.yaml [hunk 6]
```diff
@@ -282,342 +247,420 @@ components:
...
-        - id
-        - actor
-        - verb
-        - object
+        - course
+    CMI5.Course:
+      title: null
+      type: object
+      description: |-
+        Courde definition as based on cmi5 course definition:
```
Comment (hhaarsma) [inline reference/course.v1.yaml:360]: Typo - should be Course

File: reference/course.v1.yaml [hunk 7]
```diff
@@ -282,342 +247,420 @@ components:
...
+        - description
+      description: |-
+        A block describes a chapter or paragraph of a course. It describes title, additional description, links to objectives and the assignable learning units (au) that are part of this chapter or paragraph.
+        The structure is recursive, so each block can contain other blocks. In practice the number of levels in secondary education will be 2: chapters, followed by paragraphs, containing au's.
+
+        the description field can be used to specify more information on the chapter or paragraph, for example filling it with learning goals so the teacher and student can see the bigger picture.
+
+        there are 3 additional, optional fields: 
+        * instructions : additional instructions for the student, if any
+        * studyload indicator: an integer that gives a relative studyload, with this you can specify that this block is expected to take a factor 2 or 3 more time to complete. this is a great help for planning. It is reative, not an absolute time in hours or days
```
Comment (hhaarsma) [inline reference/course.v1.yaml:526]: Typo: reative instead of relative

File: reference/progress.v1.yaml [hunk 8]
```diff
@@ -0,0 +1,670 @@
...
+      type: object
+      properties:
+        id:
+          type: string
+        activityId:
+          type: string
+        studentEckId:
+          type: string
+          description: The eckid of the student for which the progress is reported.
+        studentId:
```
Comment (hhaarsma) [inline reference/progress.v1.yaml:633]: Wondering which value would be expected for studentId - I guess the Learning Application internal studentId doesnt add any value for other parties. Isnt the studentEckId sufficient?

**Review by hhaarsma — COMMENTED** [review]
Nice! Some smaller remarks added to the files.

One question that came to my mind, do we want to explicitly document the CMI5 standard and the extension (part is also in the XSDs). I see there is also added value in adding it to the documentation.

File: reference/progress.v1.yaml (hunk 8 above)
Comment (mcginkel) [reply reference/progress.v1.yaml:633]: probably : yes. The student id would be the same as the id recieved from the SIS API. Until ECKId is unique available this is still a double identifier

Comment (edwinverwoerd) [reply reference/progress.v1.yaml:633]: Why did we not made this attribute smart like studentId . Now you have to look at the context of the request or result.

Can we please change within the API's the id that means this is a student to StudentId !

**Pull request merged**